external/storm-druid/src/main/java/org/apache/storm/druid/bolt/DruidBeamBolt.java

```diff
@@ -0,0 +1,110 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.storm.druid.bolt;
+
+import com.metamx.tranquility.tranquilizer.MessageDroppedException;
+import com.metamx.tranquility.tranquilizer.Tranquilizer;
+import com.twitter.util.Future;
+import com.twitter.util.FutureEventListener;
+import org.apache.storm.task.OutputCollector;
+import org.apache.storm.task.TopologyContext;
+import org.apache.storm.topology.OutputFieldsDeclarer;
+import org.apache.storm.topology.base.BaseRichBolt;
+import org.apache.storm.tuple.Fields;
+import org.apache.storm.tuple.Tuple;
+import org.apache.storm.tuple.Values;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+/**
+ * Basic bolt implementation for storing data to Druid datastore.
+ * <p/>
+ * This implementation uses Druid's Tranquility library (https://github.com/druid-io/tranquility)
+ * to send to druid store.
+ * Some of the concepts are borrowed from Tranquility storm connector implementation.
+ * (https://github.com/druid-io/tranquility/blob/master/docs/storm.md)
+ *
+ * By default this Bolt expects to receive tuples in which "event" field gives your event type.
+ * This logic can be changed by implementing ITupleDruidEventMapper interface.
+ * <p/>
+ *
+ */
+public class DruidBeamBolt<E> extends BaseRichBolt {
+
+    private volatile  OutputCollector collector;
+    private DruidBeamFactory<E> beamFactory = null;
+    private DruidConfig druidConfig = null;
+    private Tranquilizer<E> tranquilizer = null;
+    private ITupleDruidEventMapper<E> druidEventMapper = null;
+
+    public DruidBeamBolt(DruidBeamFactory<E> beamFactory, ITupleDruidEventMapper<E> druidEventMapper, DruidConfig druidConfig) {
+        this.beamFactory = beamFactory;
+        this.druidConfig = druidConfig;
+        this.druidEventMapper = druidEventMapper;
+    }
+
+    @Override
+    public void prepare(Map stormConf, TopologyContext context, OutputCollector collector) {
+        this.collector = collector;
+        tranquilizer = Tranquilizer.builder()
+                .maxBatchSize(druidConfig.getMaxBatchSize())
+                .maxPendingBatches(druidConfig.getMaxPendingBatches())
+                .lingerMillis(druidConfig.getLingerMillis())
+                .blockOnFull(druidConfig.isBlockOnFull())
+                .build(beamFactory.makeBeam(stormConf, context));
+        this.tranquilizer.start();
+    }
+
+    @Override
+    public void execute(final Tuple tuple) {
+      Future future = tranquilizer.send((druidEventMapper.getEvent(tuple)));
+      future.addEventListener(new FutureEventListener() {
+          @Override
+          public void onFailure(Throwable cause) {
+              if (cause instanceof MessageDroppedException) {
+                  collector.ack(tuple);
+                  if (druidConfig.isEnableDiscardStream())
```

Comment:
@satishd  Since this is internal implementation, I decided to use existing boolean variable.  I felt boolean check is better than null check.  Updated the PR.